### ISSUE TITLE
Details about hyphenation in Adlam

### ISSUE THREAD

**Issue by r12a** [w3c/afrlreq#7]
From https://github.com/w3c/afrlreq/issues/2#issuecomment-664089616 we learned that 

> In Adlam, words longer than 2 syllables can be broken using hyphenation.

On the Web, paragraph widths and the content of lines is always changing due to the width of the device used, or the sizing of a window.  This means that automatic hyphenation is particularly useful, so that the line-breaking can adapt to handle whatever word appears at the line end.

 Is there any Web browser that does automatic hyphenation of Adlam script text?

How much of an issue would you say it is for Fula authors not to have automatic hyphenation? Does it significantly affect everyday authoring and reading on the Web?  Or is it mostly an issue for advanced typographic layout, such as  is found in carefully typeset magazines, ebooks, etc?

Also, what are the potential break-points for hyphenation, besides the 2 syllable rule. For example:

a. can a word be hyphenated before any letter that won't fit on the line, eg. 
![Screenshot 2020-07-27 at 15 01 46](https://user-images.githubusercontent.com/4839211/88550962-32e7ca80-d01a-11ea-843c-3e0e495e57ea.png)

b. do the break-points have to occur between syllable boundaries, and can they appear before any syllable boundary?
![Screenshot 2020-07-27 at 15 02 03](https://user-images.githubusercontent.com/4839211/88550991-3d09c900-d01a-11ea-84e3-afd335dad8b3.png)

c. do the break-points only occur at morphological boundaries, or following special orthographic rules?

**Comment by r12a** [comment]
Another question about hyphenation, arising from the example at https://github.com/w3c/afrlreq/issues/5#issuecomment-664084563 ...

When a word is hyphenated, is it preferable to:

A. drop the cursive joining around the hyphen (as shown in https://github.com/w3c/afrlreq/issues/5#issuecomment-664084563), eg. 

![Screenshot 2020-07-27 at 15 20 26](https://user-images.githubusercontent.com/4839211/88553064-c4f0d280-d01c-11ea-9fd0-92a04b0723ce.png)


B. maintain the cursive joining, eg. 

![Screenshot 2020-07-27 at 15 20 15](https://user-images.githubusercontent.com/4839211/88553126-d508b200-d01c-11ea-9d8d-fa85e5b97dc0.png)

(The [CSS spec](https://drafts.csswg.org/css-text-3/#example-e625ec40) currently says that for Arabic-script text (ie. Uighur, because Arabic language text is not hyphenated), the joining should be maintained.)

**Comment by NeilSureshPatel** [comment]
I did quick check of Safari, FireFox, Chrome and Vivaldi on MacOSX and none of them automatically hyphenate Adlam text.

**Comment by r12a** [comment]
Yep. I'm not surprised by that. [Here's a test rig](https://w3c.github.io/i18n-tests/exploratory/hyphenation/int-hyphens?lang=fuf&fontSize=46&fontFamily=&width=378&hyphens=auto&text=%F0%9E%A4%89%F0%9E%A4%A9%F0%9E%A4%AB%20%F0%9E%A4%B2%F0%9E%A5%8B%F0%9E%A4%BA%F0%9E%A4%AE%F0%9E%A5%85%F0%9E%A4%A3%F0%9E%A4%AD%20%F0%9E%A4%A5%F0%9E%A4%AD%F0%9E%A5%85%F0%9E%A4%B6%F0%9E%A4%AE%20%F0%9E%A4%AB%20%F0%9E%A4%B8%F0%9E%A4%A2%F0%9E%A4%B3%F0%9E%A5%86%F0%9E%A4%AD%F0%9E%A4%A4%F0%9E%A4%A2%F0%9E%A4%B2%F0%9E%A4%BC%F0%9E%A4%A2%F0%9E%A5%84%F0%9E%A4%BA%F0%9E%A4%A2%F0%9E%A4%A4%20%F0%9E%A4%AB%F0%9E%A4%BC%F0%9E%A4%AB%20%F0%9E%A4%AB%F0%9E%A4%A9%F0%9E%A4%AB%20%F0%9E%A4%A8%F0%9E%A4%AE%F0%9E%A4%BC%F0%9E%A4%AD%20%F0%9E%A4%B8%F0%9E%A4%B5%F0%9E%A5%85%F0%9E%A4%AC%F0%9E%A4%AE%20%F0%9E%A4%B2%F0%9E%A5%8B%F0%9E%A4%A3%F0%9E%A4%AD%F0%9E%A4%AA%F0%9E%A4%A3%F0%9E%A4%AB%20%F0%9E%A4%AB%20%F0%9E%A4%B2%F0%9E%A5%8B%F0%9E%A4%A3%F0%9E%A4%AB%F0%9E%A4%AA%20%F0%9E%A4%A9%20%F0%9E%A4%AD%F0%9E%A4%B4%F0%9E%A4%B2%F0%9E%A4%BA%F0%9E%A4%B5%F0%9E%A4%B4%F0%9E%A4%B5%F0%9E%A4%A5%F0%9E%A5%86%F0%9E%A4%A2%F0%9E%A5%84%F0%9E%A4%BA%F0%9E%A4%B5.) that i'll point to when i write this stuff up.  (Use the slider to adjust the test box width and look for wrapping.)

**Comment by NeilSureshPatel** [comment]
What is the difference between the right and left columns? The right one seems to have wider word-spaces and a different period despite being the same font as the left.

**Comment by r12a** [comment]
Left box is ordinary HTML. Right box is a textarea.  There are often different renderings for one or the other.  (/me takes a note to make that - and some other things - clearer on the UI...)

**Comment by r12a** [comment]
The differences in this case seem to arise from missing punctuation in the Noto font (space and period).  They fall back to different fonts in the HTML and the text area.  Here's [the same test](https://w3c.github.io/i18n-tests/exploratory/hyphenation/int-hyphens?lang=fuf&fontSize=46&fontFamily=Noto%20Sans%20Adlam%20New&width=378&hyphens=auto&text=%F0%9E%A4%89%F0%9E%A4%A9%F0%9E%A4%AB%20%F0%9E%A4%B2%F0%9E%A5%8B%F0%9E%A4%BA%F0%9E%A4%AE%F0%9E%A5%85%F0%9E%A4%A3%F0%9E%A4%AD%20%F0%9E%A4%A5%F0%9E%A4%AD%F0%9E%A5%85%F0%9E%A4%B6%F0%9E%A4%AE%20%F0%9E%A4%AB%20%F0%9E%A4%B8%F0%9E%A4%A2%F0%9E%A4%B3%F0%9E%A5%86%F0%9E%A4%AD%F0%9E%A4%A4%F0%9E%A4%A2%F0%9E%A4%B2%F0%9E%A4%BC%F0%9E%A4%A2%F0%9E%A5%84%F0%9E%A4%BA%F0%9E%A4%A2%F0%9E%A4%A4%20%F0%9E%A4%AB%F0%9E%A4%BC%F0%9E%A4%AB%20%F0%9E%A4%AB%F0%9E%A4%A9%F0%9E%A4%AB%20%F0%9E%A4%A8%F0%9E%A4%AE%F0%9E%A4%BC%F0%9E%A4%AD%20%F0%9E%A4%B8%F0%9E%A4%B5%F0%9E%A5%85%F0%9E%A4%AC%F0%9E%A4%AE%20%F0%9E%A4%B2%F0%9E%A5%8B%F0%9E%A4%A3%F0%9E%A4%AD%F0%9E%A4%AA%F0%9E%A4%A3%F0%9E%A4%AB%20%F0%9E%A4%AB%20%F0%9E%A4%B2%F0%9E%A5%8B%F0%9E%A4%A3%F0%9E%A4%AB%F0%9E%A4%AA%20%F0%9E%A4%A9%20%F0%9E%A4%AD%F0%9E%A4%B4%F0%9E%A4%B2%F0%9E%A4%BA%F0%9E%A4%B5%F0%9E%A4%B4%F0%9E%A4%B5%F0%9E%A4%A5%F0%9E%A5%86%F0%9E%A4%A2%F0%9E%A5%84%F0%9E%A4%BA%F0%9E%A4%B5) but with the font set to Noto Sans Adlam New – which removes the difference.  (As long as you have that font installed, of course.)

**Comment by NeilSureshPatel** [comment]
Ah, of course. The standard Noto fonts don't include common punctuation. We have included them in our Noto builds (Noto Sans Adlam New) and I believe that have convinced Google to do the same for V3 Noto Sans Adlam.
